FakebookNotifications/FakebookNotifications.Testing/IntegrationTests/NotificationControllerTest.cs

```diff
@@ -0,0 +1,188 @@
+using System.Threading.Tasks;
+using Xunit;
+
+namespace FakebookNotifications.Testing.IntegrationTests
+{
+    public class NotificationControllerTest
+    {
+
+        private static Domain.Models.Notification GetDummyNotification(string type)
+        {
+            return new()
+            {
+                Id = "0123",
+                Type = new(type, 1),
+                LoggedInUserId = "test@test.com",
+                TriggerUserId = "test@test.com",
+                HasBeenRead = false,
+                Date = new DateTime(2021, 04, 05)
+            };
+        }
+
+        private readonly Domain.Models.User testUser1 = new()
+        {
+            Id = "01",
+            Connections = new List<string>{
+                "00", "02", "03"
+            },
+            Email = "test@test.com"
+        };
+
+        private readonly List<string> groupIds = new()
+        {
+            "test@test.com",
+            "group1",
+            "group2",
+            "group3"
+        };
+
+        private readonly List<string> clientIds = new() { "00", "01", "02", "03", "04", "05" };
+
+        private readonly Mock<IHubCallerClients> mockClients = new();
+        private readonly Mock<IGroupManager> mockGroups = new();
+        private readonly Mock<IClientProxy> mockClientProxy = new();
+        private readonly Mock<HubCallerContext> mockContext = new();
+        private readonly NotificationHub mockHub;
+
+        private readonly UserRepo _userRepo;
+        private readonly NotificationsRepo _noteRepo;
+        private readonly NotificationsDatabaseSettings settings;
+        private readonly NullLogger<NotificationsContext> _logger;
+
+        public NotificationControllerTest()
+        {
+            //mocking signalr elements for tests
+            mockClients.Setup(client => client.All).Returns(mockClientProxy.Object);
+            mockClients.Setup(client => client.Group(groupIds[0])).Returns(mockClientProxy.Object);
+            mockClients.Setup(client => client.Caller).Returns(mockClientProxy.Object);
+            mockClients.Setup(client => client.OthersInGroup(It.IsIn<string>(groupIds))).Returns(mockClientProxy.Object);
+            mockGroups.Setup(group => group.AddToGroupAsync(It.IsIn<string>(clientIds), It.IsIn<string>(groupIds), new CancellationToken())).Returns(Task.FromResult(true));
+            mockGroups.Setup(group => group.RemoveFromGroupAsync(It.IsIn<string>(clientIds), It.IsIn<string>(groupIds), new CancellationToken())).Returns(Task.FromResult(true));
+            mockContext.Setup(context => context.ConnectionId).Returns("01");
+            
+
+            _logger = new NullLogger<NotificationsContext>();
+            Mock<IOptions<NotificationsDatabaseSettings>> mockSettings = new();
+            settings = new NotificationsDatabaseSettings
+            {
+                ConnectionString = "mongodb+srv://ryan:1234@fakebook.r8oce.mongodb.net/Notifications?retryWrites=true&w=majority",
```

Comment:
don't send your database connection string to github, even your test database. remove this string & change the password on it.
if we want a test database, better to run it with docker in the pipeline. we'd want a "trait" on all tests that required that, so we could skip running those tests when we don't want to run the test db.
but you're confused on what you're doing here. if you're mocking the mongodb driver, writing a unit test, then there is zero purpose to having a real database. if you're writing an integration test with a real db, then there's no purpose to mocking mongodb driver. figure out what kind of tests you want, and if you want both kinds, put them in separate namespaces at least.